### DISCUSSION TITLE
Serve a Uno-App from an ASP.Net Core-App

### DISCUSSION THREAD

**Discussion by TopperDEL** [unoplatform/uno#16855] [Q&A]
I have an existing ASP.Net Core WebAPI where I now want to serve a Uno-App from a specific url. This should be possible, if I understand it right.

So I have my WebAPI-project. I would create a Uno-app in another folder next to it and add the Uno-Head to my existing solution. How could I now say "serve my Uno-app from www.mywebapi.com/myUnoApp/". Is this possible?

**Comment by jeromelaban** [comment]
You can use the `server` option in the template: https://platform.uno/docs/articles/getting-started/wizard/using-wizard.html#server.

**Reply by TopperDEL** [reply]
But this would create a new WebAPI-project, right?

**Reply by TopperDEL** [reply]
Oh wait, I see what you mean. I should be able to add this to my existing WebAPI-project. Will try...

**Reply by TopperDEL** [reply]
Hm, I cannot add an existing Uno-App to my solution or to a newly created empty solution. "The provided SDK 'Uno.Sdk' hasn't been found".

**Reply by TopperDEL** [reply]
Ah, global.json needs to be there. Obviously. It's getting late here... :)

**Reply by jeromelaban** [reply]
That's right for now that's the configuration, but it can be altered to work differently. You'll also need to configure the app this way: https://platform.uno/docs/articles/external/uno.wasm.bootstrap/doc/features-deep-linking.html.

**Reply by TopperDEL** [reply]
It's working like a charm!

So basically I had to do these steps to add the Uno-App to my existing WebAPI-project:
1. Add a global.json to the root-folder with this content:
```
{
  "msbuild-sdks": {
    "Uno.Sdk": "5.2.132"
  }
}
```
2. Add the following to `Directory.Packages.props`:
```
<ItemGroup Condition="$(UsingUnoSdk) != 'true'">
    <PackageVersion Include="Uno.Wasm.Bootstrap.Server" Version="8.0.14" />
  </ItemGroup>
```
3. Add the Uno-Head to the existing WebAPI-solution.
4. Add a project reference to the existing WebAPI-project referencing the Uno-Head. The reference has to be adjusted in the *.csproj like this:
```
<ItemGroup>
    <ProjectReference Include="..\UnoServerTest\UnoServerTest.csproj"
                      SetTargetFramework="TargetFramework=net8.0-browserwasm"
                      ReferenceOutputAssembly="false" />
  </ItemGroup>
```
5. Setup the webAPI-project to serve Uno in the `Program.cs`:
```
app.UseUnoFrameworkFiles();
app.MapFallbackToFile("index.html");
```

What I do not know yet: how could I define a "folder" for my app. Currently it is served if no other route is being hit (hence the MapFallback). And it would be nice to know if I could serve multiple Uno-projects from one WebAPI-project (like `/Admin/` is one, `/Web/` is another and so on).
